cmd/nodepool/kubevirt/create.go

```diff
@@ -167,12 +200,43 @@ func (o *ValidatedKubevirtPlatformCreateOptions) Complete() (*KubevirtPlatformCr
 		})
 	}
 
+	var hostDevices []hyperv1.KubevirtHostDevice
+	if len(o.HostDevices) > 0 {
+		for _, hostDevice := range o.HostDevices {
+			var kubevirtHostDevice hyperv1.KubevirtHostDevice
+			split := strings.Split(hostDevice, ",")
+			if len(split) > 2 {
+				panic(fmt.Sprintf("invalid KubeVirt host device setting [%s]", hostDevice))
+			}
+			if len(split) == 2 {
+				// count is being used
+				countSplit := strings.Split(split[1], ":")
+				count, err := strconv.Atoi(countSplit[1])
+				if err != nil {
+					panic(fmt.Sprintf("could not parse host device count: [%s]", hostDevice))
+				}
+				kubevirtHostDevice = hyperv1.KubevirtHostDevice{
+					DeviceName: split[0],
+					Count:      count,
+				}
+			}
+			if len(split) == 1 {
+				// count is not being used
+				kubevirtHostDevice = hyperv1.KubevirtHostDevice{
+					DeviceName: split[0],
+				}
+			}
+			hostDevices = append(hostDevices, kubevirtHostDevice)
+		}
+	}
```

Comment:
I think this needs better error handling. As it's written here, someone could attempt to use an option not named `count` and we'd parse it like it was named count. There's also no verification that the count option has a value associated with it. We blindly try to access countSplit[1] without verifying the length of countSplit. 


Here's an example of how those two issues could be resolved. 

```
		for _, hostDevice := range o.HostDevices {
			split := strings.Split(hostDevice, ",")

			kubevirtHostDevice := hyperv1.KubevirtHostDevice{
				DeviceName: split[0],
			}

                        if len(spilt) == 1 {
                            continue
			} else if len(split) > 2 {
			    panic(fmt.Sprintf("invalid KubeVirt host device setting [%s]", hostDevice))
			}

		        // parse options (count is the only supported option right now)
			countSplit := strings.Split(split[1], ":")
                        if countSplit[0] != "count" || len countSplit != 2 {
                            panic(fmt.Sprintf("invalid KubeVirt host device setting [%s]", hostDevice))
                        }
			count, err := strconv.Atoi(countSplit[1])
			if err != nil {
			    panic(fmt.Sprintf("could not parse host device count: [%s]", hostDevice))
			}
			kubevirtHostDevice. Count = count
			
			hostDevices = append(hostDevices, kubevirtHostDevice)
		}
```

Reply:
in the ` Validate()` method we're validating the argument passed with `--host-device-name` is of the correct format.
I separated the validation logic into `Validate()` and parsing and structuring logic into `Complete()`.
wouldn't that be enough to ensure only valid input be processed?
https://github.com/openshift/hypershift/pull/4101/files#diff-e9817ce7ac10df78b9412ce6725d2974cbb7f9262259f494fe74d7e0c987a69fR119-R140

Reply:
> I separated the validation logic into Validate() and parsing and structuring logic into Complete()

both the Validate and Complete logic are kind of "half" parsing this arg in different ways.

I'd recommend consolidating this parsing logic all to one function.  You can call that function in both places if you really want, but i wouldn't parse this arg twice